sensor/kubernetes/listener/resources/virtualmachine/dispatcher/virtualmachines.go

```diff
@@ -0,0 +1,68 @@
+package dispatcher
+
+import (
+	"github.com/stackrox/rox/generated/internalapi/central"
+	virtualMachineV1 "github.com/stackrox/rox/generated/internalapi/virtualmachine/v1"
+	"github.com/stackrox/rox/sensor/kubernetes/eventpipeline/component"
+	"github.com/stackrox/rox/sensor/kubernetes/listener/resources/virtualmachine/store"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	kubeVirtV1 "kubevirt.io/api/core/v1"
+)
+
+type VirtualMachineDispatcher struct {
+	clusterID string
+	store     virtualMachineStore
+}
+
+func NewVirtualMachineDispatcher(clusterID string, store virtualMachineStore) *VirtualMachineDispatcher {
+	return &VirtualMachineDispatcher{
+		clusterID: clusterID,
+		store:     store,
+	}
+}
+
+func (d *VirtualMachineDispatcher) ProcessEvent(
+	obj interface{},
+	_ interface{},
+	action central.ResourceAction,
+) *component.ResourceEvent {
+	unstructuredObj, ok := obj.(*unstructured.Unstructured)
+	if !ok {
+		log.Errorf("not of type 'Unstructured': %T", obj)
+		return nil
+	}
+	virtualMachine := &kubeVirtV1.VirtualMachine{}
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(unstructuredObj.Object, virtualMachine); err != nil {
+		log.Errorf("unable to convert 'Unstructured' to 'VirtualMachine': %v", err)
+		return nil
+	}
+	if virtualMachine.GetUID() == "" {
+		log.Errorf("convertion from unstructured failed: %v", obj)
```

Comment:
```suggestion
		log.Errorf("conversion from unstructured failed: %v", obj)
```
but please maybe apply my matching suggestions from `virtualmachineinstances.go`.